game/world/objects/interactive.h

```diff
@@ -151,6 +151,7 @@ class Interactive : public Vob {
     int                          stepsCount = 0;
 
     int32_t                      state         = -1;
+    bool                         wait          = false;
```

Comment:
Can it be moved to PlayerControl ? Having this for every interactive object is unnecessary

Reply:
I could use a variable that already exists like `isLockCracked`. It even does the same just for chests and it should be safe since interactive can't be chest and ladder at the same time.

Reply:
Idea is to have cleaner code by keeping player interaction stuff in `PlayerControl` and common stuff (npc and player) in `Interactive`.

Reply:
I just don't see how. For chest 
```
  if(needToLockpick(npc)) {
      ...
      return; // chest is locked - need to crack lock first
      }

```

is checked to prevent playing the chest opening animation too early. The same is needed for ladder to check if player has up or down key pressed.

Reply:
> Is this intended?

No, it's not. There is keyPress - ketRepeat - keyRelease workflow, should work same for any key-type.